src/sysext/sysext.c

```diff
@@ -1187,6 +1188,7 @@ static int mount_overlayfs_with_op(
 
         int r;
         const char *top_layer = NULL;
+        int atfd = -1;
```

Comment:
use `_cleanup_close_` here. And we also use `-EBADF` as value for "uninitialized fd".